src/components/common/adblock_detector.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { HTMLAttributes } from 'react';
+
+interface Props extends HTMLAttributes<HTMLButtonElement> {}
+
+interface State {}
+
+class AdBlockDetector extends React.Component<Props, State> {
```

Comment:
Just to save a couple of lines, you can set this to be:
```typescript
class AdBlockDetector extends React.Component<Props, {}> {
```
... and remove line 5.